recipes/NAS/orion_search.py

```diff
@@ -0,0 +1,564 @@
+            lr = sum(lrl) / len(lrl)
+
+            # world_size = torch.distributed.get_world_size()
+
+            logging.info(
+                "Train: {} [{:>4d}/{} ({:>3.0f}%)]  "
+                "Loss: {loss.val:>9.6f} ({loss.avg:>6.4f})  "
+                "Time: {batch_time.val:.3f}s  "
+                "({batch_time.avg:.3f}s)  "
+                "LR: {lr:.3e}  "
+                "Data: {data_time.val:.3f} ({data_time.avg:.3f})".format(
+                    epoch,
+                    batch_idx,
+                    len(loader),
+                    100.0 * batch_idx / last_idx,
+                    loss=losses_m,
+                    batch_time=batch_time_m,
+                    lr=lr,
+                    data_time=data_time_m,
+                )
+            )
+
+        end = time.time()
+        # end for
+
+    if hasattr(optimizer, "sync_lookahead"):
+        optimizer.sync_lookahead()
+
+    return OrderedDict([("loss", losses_m.avg)])
+
+
+def validate(model, loader, loss_fn, log_suffix=""):
+    batch_time_m = AverageMeter()
+    losses_m = AverageMeter()
+    prec1_m = AverageMeter()
+    prec5_m = AverageMeter()
+
+    model.eval()
+    log_interval = 50
+    end = time.time()
+    last_idx = len(loader) - 1
+    with torch.no_grad():
+        for batch_idx, (input, target) in enumerate(loader):
+            last_batch = batch_idx == last_idx
+
+            output = model(input)
+            if isinstance(output, (tuple, list)):
+                output = output[0]
+
+            loss = loss_fn(output, target)
+            prec1, prec5 = accuracy(output, target, topk=(1, 5))
+
+            reduced_loss = loss.data
+
+            torch.cuda.synchronize()
+
+            losses_m.update(reduced_loss.item(), input.size(0))
+            prec1_m.update(prec1.item(), output.size(0))
+            prec5_m.update(prec5.item(), output.size(0))
+
+            batch_time_m.update(time.time() - end)
+
+            if last_batch or batch_idx % log_interval == 0:
+                log_name = "Test" + log_suffix
+                logging.info(
+                    "{0}: [{1:>4d}/{2}]  "
+                    "Time: {batch_time.val:.3f} ({batch_time.avg:.3f})  "
+                    "Loss: {loss.val:>7.4f} ({loss.avg:>6.4f})  "
+                    "Prec@1: {top1.val:>7.4f} ({top1.avg:>7.4f})  "
+                    "Prec@5: {top5.val:>7.4f} ({top5.avg:>7.4f})".format(
+                        log_name,
+                        batch_idx,
+                        last_idx,
+                        batch_time=batch_time_m,
+                        loss=losses_m,
+                        top1=prec1_m,
+                        top5=prec5_m,
+                    )
+                )
+            end = time.time()
+
+    metrics = OrderedDict(
+        [("loss", losses_m.avg), ("prec1", prec1_m.avg), ("prec5", prec5_m.avg)]
+    )
+    return metrics
+
+
+def objective(alpha, beta, B0, t_zero, res, epochs):
```

Comment:
let's make the objective method only related to the objective function, meaning the cost function (ACC / complexity tradeoff) -- this should be a simple composition of functions of the ACC and the complexity; break this function into multiple methods and simplify.